File: Python/30temperatureConverter.py
# 30. - com.mouredev.weeklychallenge2022 #42
""" Crea una función que transforme grados Celsius en Fahrenheit y viceversa.
Para que un dato de entrada sea correcto debe poseer un símbolo "°" y su unidad ("C" o "F"). En caso contrario retornará un error. """


def temperatureConverter(degrees):
    try:
        if "ºC" in degrees:
            celsiusDegrees = degrees.replace(" ","").replace("ºC", "")
            
            if celsiusDegrees is not None:
                fahrenheit = int(celsiusDegrees) * (9 / 5) + 32
                return str(round(fahrenheit,2))+" ºF"
            
        elif "ºF" in degrees:
            fahrenheitDegrees = degrees.replace(" ","").replace("ºF", "")

            if fahrenheitDegrees is not None:
                celsius = (int(fahrenheitDegrees) - 32) * 5 / 9 
                return str(round(celsius,2))+" ºC"
    except:
        return None


print(temperatureConverter("100ºC"))
print(temperatureConverter("100ºF"))
print(temperatureConverter("100C"))
print(temperatureConverter("100"))
print(temperatureConverter("- 100ºC"))
print(temperatureConverter("- 100ºF"))
print(temperatureConverter("100AºC"))
print(temperatureConverter("ºC"))

File: Java/30temperatureConverter.java
package questions;

public class Solution {
	
	
    /* # 30. - com.mouredev.weeklychallenge2022 #42 */
    /* Crea una función que transforme grados Celsius en Fahrenheit y viceversa.
    Para que un dato de entrada sea correcto debe poseer un símbolo "°" y su unidad ("C" o "F"). En caso contrario retornará un error. */
	
	
	public String temperatureConveter(String degrees) {
		try {
			if (degrees.contains("ºC")) {
				String celsiusDegrees = degrees.replace(" ", "").replace("ºC","");
				if (celsiusDegrees != null) {
					float celsiusDegreesNum = Float.parseFloat(celsiusDegrees);
					float fahrenheit = (float) (celsiusDegreesNum * (9.0 / 5.0) + 32);
					String fahrenheitString = Float.toString(fahrenheit);
					fahrenheitString = fahrenheitString + "ºF";
					return fahrenheitString;
				}		
			} else if (degrees.contains("ºF")) {
				String fahrenheitDegrees = degrees.replace(" ", "").replace("ºF","");
				if (fahrenheitDegrees != null) {
					float fahrenheitDegreesNum = Float.parseFloat(fahrenheitDegrees);
					float celsius = (float) ((fahrenheitDegreesNum - 32) * (5.0/9.0));
					String celsiusString = Float.toString(celsius);
					celsiusString = celsiusString + "ºC";
					return celsiusString;
				}	
			}
		} catch(Exception e) {
			return null;
		}
		return null;
	}


    System.out.println(solution.temperatureConveter("100ºC"));
    System.out.println(solution.temperatureConveter("100ºF"));
    System.out.println(solution.temperatureConveter("100C"));
    System.out.println(solution.temperatureConveter("100"));
    System.out.println(solution.temperatureConveter("- 100ºC"));
    System.out.println(solution.temperatureConveter("- 100ºF"));
    System.out.println(solution.temperatureConveter("100AºC"));
    System.out.println(solution.temperatureConveter("ºC"));
}
